src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -256,6 +257,24 @@ private function getNormalizer(mixed $data, ?string $format, array $context): ?N
                     continue;
                 }
 
+                if ($normalizer instanceof SupportedTypesMethodInterface && (null !== $supportedTypes = $normalizer->getSupportedTypes())) {
+                    foreach ($supportedTypes as $supportedType => $isCacheable) {
+                        if ($type !== $supportedType && !is_subclass_of($type, $supportedType, true)) {
+                            continue;
+                        }
+
+                        if ($isCacheable && $normalizer->supportsNormalization($data, $format, $context)) {
+                            $this->normalizerCache[$format][$type][$k] = true;
```

Comment:
Why still a cache per `$format` when `->getSupportedTypes()` is not format-aware? Every format would have the same cache. Or is this for the BC-layer?